views/project/src/components/EDATargetCheckbox.vue

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+import type {
+  YosysOptions,
+  NextpnrOptions
+} from 'edacation';
+import { defineComponent } from 'vue';
+
+import { state as globalState } from '../state';
+
+export default defineComponent({
+  props: {
+    worker: {
+      type: String,
+      required: true
+    },
+    configTarget: {
+      type: String,
+      required: true
+    },
+    label: {
+      type: String
+    }
+  },
+  data() {
+    return {
+      state: globalState,
+      option: false
+    };
+  },
+  methods: {
+    toggleOption() {
+      this.option = !this.option;
+
+      const { project } = this.state;
+      if (!project) return;
+
+      project.configuration = project.configuration || {};
+      project.configuration.defaults = project.configuration.defaults || {};
+
+      if (this.worker === 'yosys') {
```

Comment:
You can use the value of `worker` to access the config, which removes the need for this `if`-statement. For example: `project.configuration.defaults[this.worker].options`.